src/transformers/modeling_utils.py

```diff
@@ -611,7 +612,14 @@ def _load_state_dict_into_meta_model(
         # We convert floating dtypes to the `dtype` passed. We want to keep the buffers/params
         # in int/uint/bool and not cast them.
         if dtype is not None and torch.is_floating_point(param):
-            param = param.to(dtype)
+            if (
+                keep_in_fp32_modules is not None
+                and any(module_to_keep_in_fp32 in param_name for module_to_keep_in_fp32 in keep_in_fp32_modules)
+                and dtype != torch.bfloat16
```

Comment:
```suggestion
                and dtype == torch.float16
```
This is specifically for float16 dtypes (I don't think we get the int8 here?)

Reply:
Yes, totally!